grails-app/services/org/broadinstitute/orsp/UserService.groovy

```diff
@@ -151,4 +152,48 @@ class UserService {
         }
     }
 
+    /**
+     * Edits a user's roles
+     *
+     * @param userId        The user's id
+     * @param rolesToAssign Roles String array to be assigned
+     */
+    void editUserRoles (Integer userId, ArrayList<String> rolesToAssign) {
+        if (userId != null) {
+            User user = User.findById(userId)
+            if (user != null && validateRoles(rolesToAssign)) {
+                queryService.deleteOrspUserRoles(userId)
+                if (!rolesToAssign.isEmpty()) {
+                    queryService.updateOrspUserRoles(user, rolesToAssign)
+                }
+            } else {
+                log.error("Error while trying to modify roles to userId: ${userId}.")
+                throw IllegalArgumentException()
```

Comment:
It would be nice to know what the error is here. Can you pass that into the exception so it can bubble up? Also, it would be nice to know *why* the failure happened. Was the user null? What failed in the validation? etc.

Reply:
Could you please check if the [modification](https://github.com/broadinstitute/orsp-pub/pull/178/files#diff-0329d5293b4b710e0a17b621f105f08bR207) covers this request? I think it does. Otherwise please let me know.

Reply:
👍 It does, thank you!